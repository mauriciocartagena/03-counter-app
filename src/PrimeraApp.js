import React, { Fragment } from 'react';
import PropTypes from 'prop-types';


// FC Functional Component
const PrimeraApp = ( { saludo,subtitle } ) => {



    return(
        <Fragment>
            <h1> { saludo } </h1>
            {/* <pre>{ JSON.stringify( saludo, null, 3 ) }</pre> */}
            <p> { subtitle } </p>
        </Fragment>
    );
}

PrimeraApp.propTypes = {
    saludo: PropTypes.string.isRequired
}

PrimeraApp.defaultProps={
    subtitle:'Soy un subtitulo',
}

export default PrimeraApp;