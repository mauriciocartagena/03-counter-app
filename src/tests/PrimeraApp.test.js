import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import PrimeraApp from '../PrimeraApp';

describe('Pruebas en <Primer App>', () => {
    
    // test('Debe de mostar el mensaje de "Hola, Soy Goku"', () => {
    //     const saludo = 'Hola, so Goku';
    //     const { getByText  } = render(<PrimeraApp saludo={ saludo }  ></PrimeraApp>)
    //     expect(getByText( saludo )).toBeInTheDocument();
    // })

    test('debe de mostar <PrimeraApp /> correctamente', () => {

        const saludo = 'Hola soy Goku';
        const wrapper = shallow(<PrimeraApp saludo={ saludo } />);

        expect( wrapper ).toMatchSnapshot();


    })

    test('Debe de mostar el subtitulo enviado por props', () => {

        const saludo = 'Hola soy Goku';

        const subTitulo = 'Soy un subtitulo'

        const wrapper = shallow(
            <PrimeraApp 
                saludo={ saludo }
                subtitle={ subTitulo } 
            />
        );

        const textoParrafo = wrapper.find('p').text();
        expect( textoParrafo ).toBe( subTitulo );

    })
    
    
    

});
