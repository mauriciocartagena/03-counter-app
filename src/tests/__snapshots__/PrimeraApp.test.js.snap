// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <Primer App> Debe de mostar el subtitulo enviado por props 1`] = `
<Fragment>
  <h1>
     
    Hola soy Goku
    !!!
  </h1>
  <p>
    Soy un subtitulo
  </p>
</Fragment>
`;

exports[`Pruebas en <Primer App> debe de mostar <PrimeraApp /> correctamente 1`] = `
<Fragment>
  <h1>
     
    Hola soy Goku
    !!!
  </h1>
  <p>
    Soy un subtitulo
  </p>
</Fragment>
`;
